### PR TITLE
refactor(evm): rewire router

### PR DESCRIPTION
**Pull request by technicallyty**
Closes: WORLD-623

## Overview

Fixes issues preventing router from working. Adds docs for authorizing a persona address.

## Brief Changelog
- fixes `clean` makefile command to run all steps, even if one fails
- fixes router's call to `WaitForNextTick()`, used to assume bool meant timeout, but it meant success.
- added tons of logs so we can easily trace execution of cross-shard messages.
- revert router back to using an environment variable. game shard registration would be too tricky given that the rollkit CLI is non-functional. (celesetia is aware and working on a repro + fix).
- removes some checks from the CLI command as it was failing for valid gRPC addresses.
- fixes the end to end game to match what the precompile expects.

## Testing and Verifying

i tested manually that sending cross-shard txs works, that result fetching works, and queries works.

### CODE REVIEW

File: evm/precompile/router/router.go [hunk 1]
```diff
@@ -35,22 +37,36 @@ func (c *Contract) SendMessage(
 	messageID string,
 	namespace string,
 ) (bool, error) {
+	log.Logger.Debug().Msg("inside SendMessage precompile function called")
 	pCtx := vm.UnwrapPolarContext(ctx)
-	err := c.rtr.SendMessage(ctx, namespace, pCtx.MsgSender().String(), messageID, message)
-	if err != nil {
-		return false, err
+	if c.rtr != nil {
```
Comment (jerargus) [inline evm/precompile/router/router.go:42]: nit: 
```
if c.rtr == nil {
    log.Logger.Debug()...
    return false, fmt.Error("nil router")
}
// happy path...
```

**Review by jerargus — APPROVED** [review]

File: evm/router/router.go [hunk 2]
```diff
@@ -97,52 +78,74 @@ func NewRouter(logger log.Logger, ctxGetter GetQueryCtxFn, addrGetter GetAddress
...
+func (r *routerImpl) PostBlockHook(transactions types.Transactions, receipts types.Receipts, _ types.Signer) {
+	r.logger.Info("running PostBlockHook",
+		"num_transactions", len(transactions),
+	)
+	// loop over all txs
+	for i, tx := range transactions {
+		r.logger.Info("working on transaction", "tx_hash", tx.Hash().String())
+		// we check the `To` address because the precompile msg.Sender will be the contract address,
+		// NOT the tx.Origin address, or, more formally, the EOA address that triggered the tx.
+		if txTo := tx.To(); txTo != nil {
```
[the remark this answers is not in the document]
Comment (technicallyty) [reply evm/router/router.go:96]: gonna leave this as is for now, dont wanna move too much code around as all testing was done manually

File: evm/router/queue.go [hunk 3]
```diff
@@ -1,26 +1,54 @@
...
+	return msg, ok
+}
+
+func (m *msgQueue) Remove(sender common.Address) {
+	delete(m.queue, sender)
 }
 
-func (m *msgQueue) IsSet() bool {
-	return m.msg != nil && m.namespace != ""
+func (m *msgQueue) IsSet(address common.Address) bool {
```
[the remark this answers is not in the document]
Comment (technicallyty) [reply evm/router/queue.go:38]: added mutex, unsure if polaris internals will cause concurrent use

File: evm/precompile/router/router.go (hunk 1 above)
Comment (technicallyty) [reply evm/precompile/router/router.go:42]: will return to this eventually, but gonna leave as is for now

**Review by namrapatel — APPROVED** [review]
aside from removing all the forge stuff, looks niceee

**Pull request merged**